.github/scripts/ci_tests.sh

```diff
@@ -0,0 +1,46 @@
+#! /bin/bash
+
+# Copyright (c) Microsoft. All rights reserved.
+# Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+set -o errexit # Exit if command failed.
+set -o nounset # Exit if variable not set.
+set -o pipefail # Exit if pipe failed.
+
+echo -e "::group::FreeRTOS Source"
+
+if [ ! -d "libs/FreeRTOS" ]; then
+    git clone https://github.com/FreeRTOS/FreeRTOS.git libs/FreeRTOS
+    pushd libs/FreeRTOS
+    git checkout -b c8fa483b68c6c1149c2a7a8bc1e901b38860ec9b
+    git submodule sync
+    git submodule update --init --recursive --depth=1
+    popd
+fi
+TEST_FREERTOS_SRC=`pwd`/libs/FreeRTOS
+
+function binary_exists()
```

Comment:
Nit - recommend this be something like "exit_if_binary_does_not_exist()" or something better word-smithed so the side effect of script ending is a bit more clear.  Though maybe in bash land this is common practice??